src/binary/Builtin.cc

```diff
@@ -632,6 +632,39 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  static struct tm *
+  hash_to_tm(VALUE hash)
+  {
+    struct tm res;
+
+    res.tm_sec    = rb_hash_aref(hash, ID2SYM(rb_intern("tm_sec")));
```

Comment:
You get a VALUE and treat it as an int. I think it should be: 
```
res.tm_sec = NUM2INT(rb_hash_aref(...));
```

http://ruby-doc.com/docs/ProgrammingRuby/html/ext_ruby.html#UB